hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/monitor/ContainersMonitorImpl.java

```diff
@@ -588,6 +589,31 @@ private void initializeProcessTrees(
     }
 
     /**
+     * Get the best process tree calculator.
+     * @param pId container process id
+     * @return process tree calculator
+     */
+    private ResourceCalculatorProcessTree
+        getResourceCalculatorProcessTree(String pId) {
+      ResourceCalculatorProcessTree pt = null;
+
+      // CGroups is best in perforance, so try to use it, if it is enabled
+      if (processTreeClass == null &&
```

Comment:
Impossible. CGroupsResourceCalculator relies on cgroups that are present only in node manager. ResourceCalculatorProcessTree is hadoop-yarn.common package